src/Components/Control.php

```diff
@@ -35,65 +35,40 @@
  */
 class Control extends Application\UI\Control
 {
-	/**
-	 * @persistent int
-	 */
-	public $page = 1;
+	public int $page = 1;
 
 	/**
 	 * Events
 	 *
-	 * @var array
+	 * @var mixed[]
 	 */
-	public $onShowPage;
+	public array $onShowPage;
 
-	/**
-	 * @var Utils\Paginator
-	 */
-	protected $paginator;
+	protected ?Paginator $paginator;
 
-	/**
-	 * @var string
-	 */
-	protected $templateFile;
-        
-        /**
-         * @var int 
-         */
-        protected $displayRelatedPages;
+	protected string $templateFile;
 
-	/**
-	 * @var Localization\ITranslator
-	 */
-	protected $translator;
+	protected int $displayRelatedPages;
 
-	/**
-	 * @var bool
-	 */
-	protected $useAjax = TRUE;
+	protected ?Translator $translator;
```

Comment:
Asi bych doplnil default = null, aby to pak nepadalo na te chybe s pristupem k property, ktera jeste nebyla inicializovana